arch/arm/soc/nordic_nrf5/nrf52/power.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <soc_power.h>
+#include <nrf_power.h>
+
+#if defined(CONFIG_SYS_POWER_DEEP_SLEEP)
+/* System_OFF is deepest Power state available, On exiting from this
+ * state CPU including all peripherals reset
+ */
+static void _system_off(void)
+{
+	nrf_power_system_off();
+}
+#endif
+
+void _issue_low_power_command(void)
+{
+	/* Issue _WFE Enent*/
+	__SEV();
+	__WFE();
+	__WFE();
+	if (NRF_GPIOTE->EVENTS_PORT) {
```

Comment:
Move this to line 61 in the caller of this function, and use nRFx interface if available.

Reply:
Hi @cvinayak , 


I did not find any interface available for clearing NRF_GPIOTE->EVENTS_PORT. 
Clearing the port Event is needed when sense is enabled and port event is generated. In other cases like RTC wake. reset button we will not need this, But it should not create harm in other cases.


My understanding (through reading some nrf52 spec and same applications @ https://github.com/NordicSemiconductor/nrf51-powerdown-examples) is when we are using port Event for waking up the device then we will need to clear the port event after waking. So it will be needed for IDLE ON States too. In-fact in case of System Off state device is going to reset, so we may not need this.